### PR TITLE
Switch to sub-senders

### PR DESCRIPTION
**Pull request by CodeSandwich**
Depends on https://github.com/radicle-dev/radicle-streaming/pull/24.

Removes FundingPool and switches to DaiPool's sub-senders. Adds `topUp` and `withdraw` to FundingNFT to balance lack of direct control over NFT funding streams.

### CODE REVIEW

File: src/nft.sol [hunk 1]
```diff
@@ -61,51 +66,72 @@ contract FundingNFT is ERC721, Ownable {
...
+    }
+
     function withdrawable(uint tokenId) public view returns(uint128) {
-        return pool.maxWithdraw(pool.nftID(address(this), tokenId));
+        uint128 amtPerSec = pool.getAmtPerSecSubSender(address(this), tokenId);
+        if (amtPerSec == 0) {
+            return 0;
+        }
+
+        uint128 withdrawable_ = pool.withdrawableSubSender(address(this), tokenId);
```
Comment (manuelpolzhofer) [inline src/nft.sol:111]: Can we rename it to something like?
```suggestion
        uint128 withdrawable_ = pool.amtNotStreamedSubSender(address(this), tokenId);
```

File: src/pool.sol [hunk 2]
```diff
@@ -1,53 +0,0 @@
-// SPDX-License-Identifier: GPL-3.0-only
```
Comment (manuelpolzhofer) [inline src/pool.sol:1]: 👍

**Review by manuelpolzhofer — APPROVED** [review]
Looks good. I like the new design. 
After adding some tests ready to merge. 👍

File: src/nft.sol (hunk 1 above)
Comment (CodeSandwich) [reply src/nft.sol:111]: IMO `withdrawable` is a clearer indicator of how much can be withdrawn than `amtNotStreamed`. It's also a `Pool` function, we'd need to change the whole naming convention in its API to do that.

Comment (manuelpolzhofer) [reply src/nft.sol:111]: Yes, but `withdrawable` from the pool is not the amt the NFT registry will allow you to withdraw but no strong opinion we can keep it as it

Comment (CodeSandwich) [reply src/nft.sol:111]: In both cases it tells you the highest value you can currently `withdraw` using the provided API and it behaves almost the same in `Pool` and in `FundingNFT`. I think that they SHOULD be named the same to highlight these similarities. The NFT users can't interact with (and shouldn't even be aware of) the underlying sub-senders directly anyway, so they can't confuse these `withdrawable`s, all they can see and use is the one from `FundingNFT`.

File: src/nft.sol [hunk 3]
```diff
@@ -135,4 +161,9 @@ contract FundingNFT is ERC721, Ownable {
         // test project data json
         return "QmdFspZJyihiG4jESmXC72VfkqKKHCnNSZhPsamyWujXxt";
     }
+
+    function _currLeftSecsInCycle() internal view returns(uint64) {
```
[the remark this answers is not in the document]
Comment (CodeSandwich) [reply src/nft.sol]: Fixed

**Comment by CodeSandwich** [comment]
Added `WITHDRAW_ALL` and some tests. Now only waiting on https://github.com/radicle-dev/radicle-streaming/pull/24.

**Pull request merged**